lib/Sema/ConstraintSystem.cpp

```diff
@@ -63,16 +63,44 @@ void ConstraintSystem::solveConstraints()
     // TODO: Use Result to update the best solutions for each constraint.
 }
 
+bool ConstraintSystem::applyBind(Constraint *constraint, SystemState &state)
+{
+    auto *first = constraint->getFirstType();
+    auto *second = constraint->getSecondType();
+
+    if (first == second)
+        return true;
+
+    if (llvm::isa<glu::types::TypeVariableTy>(first)
+        && !llvm::isa<glu::types::TypeVariableTy>(second)) {
+        state.typeBindings[static_cast<glu::types::TypeVariableTy *>(first)]
```

Comment:
don't use static_cast, use llvm::cast, or better, use llvm::dyn_cast in the if instead of llvm::isa to bind a variable